cmake: Switch to generated `ts_files.cmake` file

hebasto opened the pull request:
This PR:
1. Moves handling of Qt TS files into the `locale` directory.
2. Switches from inferior globbing to the explicit file list generated by the [`bitcoin-maintainer-tools/update-translations.py`](https://github.com/bitcoin-core/bitcoin-maintainer-tools/blob/main/update-translations.py) script.

Closes #32653.

hebasto committed:
cmake: Switch to generated `ts_files.cmake` file

See the `update-translations.py` script in the
`bitcoin-maintainer-tools` repository.

fanquake commented:
Can we do a translation update here, to show that this change, and https://github.com/bitcoin-core/bitcoin-maintainer-tools/pull/185, are working properly.

hebasto committed:
qt: Translations update

The translations for the following languages, which appear to be the
result of a mistake or an act of vandalism, have been discarded:
- Greek (el)
- Vietnamese (vi)

hebasto commented:
> Can we do a translation update here, to show that this change, and [bitcoin-core/bitcoin-maintainer-tools#185](https://github.com/bitcoin-core/bitcoin-maintainer-tools/pull/185), are working properly.

Sure. A commit has been added. Please see the commit message for details.

hebasto committed:
cmake, refactor: Move handling of Qt TS files into `locale` directory

This change offers a few advantages, such as:
- a more readable and cleaner `ts_files.cmake` (see the next commit);
- a scoped `ts_files` variable;
- improved code locality;
- no need to adjust the location of the resulting `*.qm` files.

BrandonOdiwuor reviewed (commented):
Code Review ACK f2bfe852236f45b9ea81d6b3f06c6a0097be3ddd

Environment: macOS 15.5, Qt 6.9.0

Build & GUI Test: Built successfully; GUI tested with generated ts_files.cmake via updated update_translations.py. No errors.
Translation Test: Kiswahili (sw) loaded correctly in GUI. Screenshot:

<img width="1098" height="559" alt="Screenshot 2025-08-02 at 09 02 04" src="https://github.com/user-attachments/assets/265a5447-be34-4eeb-94ca-2e9d37ac1cb3" />

Also tested the updated `update_transaltions.py` against master branch and confirmed `Greek (el)` and ` Vietnamese (vi)` as the only difference in `ts_files.cmake` list as intended

<img width="1104" height="373" alt="Screenshot 2025-08-02 at 09 25 32" src="https://github.com/user-attachments/assets/450e868f-732f-461e-ae4c-62a04b0ff553" />

jesterhodl commented on src/qt/locale/bitcoin_pl.ts:
```diff
@@ -95,28 +125,36 @@ Podpisywanie jest możliwe tylko z adresami typu 'legacy'.</translation>
     </message>
     <message>
         <source>Encrypt wallet</source>
         <translation type="unfinished">Zaszyfruj portfel </translation>
```
Thanks, I'll review those

fanquake replied:
Are we going to pull in the updates, or leave this as-is?

fanquake reviewed (commented):
ACK a26fbee38f95e71fbbeb6cf09e18ed7fff089ec8
